RFCs/FS-1004-result-type.md

```diff
@@ -48,3 +48,42 @@ The standard range of testing considerations for F# library types apply.
 - declare as struct
 - compiled name
 - require additional functions?
+
+### Proposal for additional functions
+
+Minimal:
+
+```fsharp
+bind : ('TSuccess -> Result<'T, 'TError>) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
+map : ('TSuccess -> 'T) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
+```
+
+Nice-to-have:
+
+```fsharp
+fold : ('TSuccess -> 'T) -> ('TError -> 'T) -> Result<'TSuccess, 'TError> -> 'T
```

Comment:
@dsyme, yes not a big difference between an explicit match, but I do find fold functions useful (and use them very frequently in my code):
```fsharp
textBox.value
|> validate
|> function 
    | Success x -> Some x
    | Error _ -> None

// or
textBox.value
|> validate
|> Result.fold Some (constant None)
```

It can also be useful to partially apply a fold function at times.

We could also provide this instead of a bunch of other proposed functions like `ofOption` or `toSeq`, so these can be defined inline as needed (which avoids some of the problems mentioned about inadvertently dropping values):

```fsharp
let ofOption = Result.fold Some (constant None)
let toSeq = Result.fold (Seq.singleton) (constant Seq.empty)
```